glean-core/src/metrics/memory_distribution.rs

```diff
@@ -304,4 +305,76 @@ impl MemoryDistributionMetric {
             test_get_num_recorded_errors(glean, self.meta(), error).unwrap_or(0)
         })
     }
+
+    /// Start a new histogram buffer associated with this timing distribution metric.
+    ///
+    /// A histogram buffer accumulates in-memory.
+    /// Data is recorded into the metric on drop.
+    pub fn start_buffer(&self) -> HistogramBuffer<'_> {
+        HistogramBuffer::new(self)
+    }
+
+    fn commit_histogram(&self, histogram: Histogram<Functional>) {
+        let metric = self.clone();
+        crate::launch_with_glean(move |glean| {
+            glean
+                .storage()
+                .record_with(glean, &metric.meta, move |old_value| {
+                    let mut hist = match old_value {
+                        Some(Metric::TimingDistribution(hist)) => hist,
+                        _ => Histogram::functional(LOG_BASE, BUCKETS_PER_MAGNITUDE),
+                    };
+
+                    hist.merge(&histogram);
+                    Metric::TimingDistribution(hist)
```

Comment:
Uh, `Metric::MemoryDistribution(hist)`, surely?

Reply:
Is this function not covered by tests?

Reply:
Oops. Should be changed.
Fun fact: doesn't matter much for the payload, at least in how the data looks.

Reply:
A test would have caught this. So there will be a test now.